packages/media/src/preloadVideoMeta.js

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const preloadVideoMeta = () => {
```

Comment:
`preload=auto` indicates that the whole video file can be downloaded, even if the user is not expected to use it. Plus, `hasVideoGotAudio` and `getFirstFrameOfVideo` modify `video.currentTime` so the video is actually forced to download. As such, I would not call this function `preloadVideoMeta` but just `preloadVideo`.

Second, contrary to `preloadImage` this merely creates a video element and doesn't  promisify the loading of the video. It doesn't even set the `src`.

To make this more DRY, we could write it more like this:

```js
function preloadVideo(src) {
	  const video = document.createElement('video');
	  video.muted = true;
	  video.crossOrigin = 'anonymous';
	  video.preload = 'auto';

	return new Promise((resolve, reject) => {
		    video.addEventListener('error', reject);
		    video.addEventListener('canplay', () => {
		    	video.currentTime = 0.99;
		    },
			    { once: true } // Important because 'canplay' can be fired hundreds of times.
		    );

		    video.addEventListener('seeked', () => resolve(video), { once: true });
		    video.src = src;
	});
}
```

Then `hasVideoGotAudio` becomes:

```js
function hasVideoGotAudio(src) {
	const video = await preloadVideo(src);
	return hasAudio(video);
```

Reply:
The name preloadVideo is used here - https://github.com/google/web-stories-wp/pull/8638/files#diff-8a1452592678d86eaca2aa587c8579a786a5c4e5b6cd1734c245cee6d57453dfR23

> function hasVideoGotAudio(src) {
	const video = await preloadVideo(src);
	return hasAudio(video);

I like this ^

Reply:
> function hasVideoGotAudio(src) {
	const video = await preloadVideo(src);
	return hasAudio(video);

How does this works for `getFirstFrameOfVideo`?

Reply:
> The name preloadVideo is used here - https://github.com/google/web-stories-wp/pull/8638/files#diff-8a1452592678d86eaca2aa587c8579a786a5c4e5b6cd1734c245cee6d57453dfR23

That function only actually loads video metadata (`preload=meta`) so _that function_ should be called `preloadVideoMetadata`. Not this one.

> How does this works for `getFirstFrameOfVideo`?

It's gonna be a bit longer because `canvas.toBlob()` needs a callback, but it really would work the same. Basically this should work:

```js
async function getFirstFrameOfVideo(src) {
  const video = await preloadVideo(src);

  return new Promise((resolve, reject) => {
    const canvas = document.createElement('canvas');
    canvas.width = video.videoWidth;
    canvas.height = video.videoHeight;

    const ctx = canvas.getContext('2d');
    try {
      ctx.drawImage(video, 0, 0, canvas.width, canvas.height);
    } catch (err) {
      // Browser probably doesn't support the video codec.
      reject(err);
    }

    canvas.toBlob(resolve, 'image/jpeg');
  });
}
```

Reply:
Need to rename the function to `preloadVideo` in this PR.

#8638 adds `preloadVideoMetadata`